### DISCUSSION TITLE
Can be "timer for mesuaremets" configurable?

### DISCUSSION THREAD

**Discussion by stevoh6** [serkri/SmartEVSE-3#116] [Ideas]
Hi, can be this hardcoded value configurable (https://github.com/serkri/SmartEVSE-3/blob/ebfc5ca66e1c779a41f8292d6928247c72e9a40c/SmartEVSE-3/src/evse.cpp#L2080)? I would like to use API for feeding current data ( and I would like to feed them every 1-5minutes wiht avg. values for selected time). Basicaly I would like to slow down "jumps" in charging speed when Im in solar or smart mode. I this it will be more robust to let charger adjusting the current (not home assistant or nodered), but with less frequency maybe add some configurale delay to adjusting current when on solar/smart mode? https://github.com/serkri/SmartEVSE-3/blob/master/SmartEVSE-3/src/evse.cpp#L792

Now after 10 seccods without API smart meter data it goes to error state (communication error). 
Maybe it is better way how to do it.

**Comment by dingo35** [comment]
Sorry that doesnt seem like a smart thing to do. We'd like to work SmartEVSE out-of-the-box with as low risk as possible; the 10s is already a long period of timr, as compared to the 1-2s intervals that are used for the modbus communications. Feel free to compile a version with your own values...

**Reply by stevoh6** [reply]
Okay, that make sense. But this is still raising some questions:

- isnt bad for the car adjusting current so often? 
- how to make smoother/how to slow the current adjustments in solar mode? Most people who has FVE and EV car has home batteries. So they havent problem with 1-3minutes supply the car charging. Which again car help not stress out solar invertor or car, or am I wrong?

Thx

**Comment by dingo35** [comment]
1. No it isnt bad for the car to adjust the current dynamically, it is what every level 2 or level 3 charger is doing. Plus, it is not the current that is adjusted, but the MAX current; the car itself decides how much it pulls.
2. In my country almost none of the people who drive EVs have a home battery. 
EV driving is promoted heavily by companies who give their people company cars, but demand it will be electric. Typically the company pays for the charger and the electricity, but not for solar panels and/or home batteries.
In short: there is no need for smoothing (maximum) charge currents. 

stevoh6 ***@***.***> schreef op 26 maart 2023 03:42:50 EDT:

**Reply by stevoh6** [reply]
Now it makes sense, thx.
